core/sr-primitives/src/generic/checked_extrinsic.rs

```diff
@@ -19,25 +19,33 @@
 
 use crate::traits::{self, Member, SimpleArithmetic, MaybeDisplay};
 use crate::weights::{Weighable, Weight};
+use crate::generic::tip::{Tip, Tippable};
 
 /// Definition of something that the external world might want to say; its
 /// existence implies that it has been checked and is good, particularly with
 /// regards to the signature.
 #[derive(PartialEq, Eq, Clone)]
 #[cfg_attr(feature = "std", derive(Debug))]
-pub struct CheckedExtrinsic<AccountId, Index, Call> {
+pub struct CheckedExtrinsic<AccountId, Index, Call, Balance> {
 	/// Who this purports to be from and the number of extrinsics have come before
 	/// from the same signer, if anyone (note this is not a signature).
 	pub signed: Option<(AccountId, Index)>,
 	/// The function that should be called.
 	pub function: Call,
+	/// An optional tip value that may or may not exist based on the underlying unchecked extrinsic.
+	///
+	/// Most often this is:
+	/// - `None` if the unchecked extrinsic does not have a tip OR it is unsigned.
+	/// - `Some` if the opposite.
+	pub tip: Option<Tip<Balance>>,
```

Comment:
should just be an additional arg in `signed`

Reply:
could be moved there but you mean probably in `unchecked_mortal_compact_tipped.rs` (god, hate this name to type it)? I will also update here but this is not really important anymore.